Replace kDP_SetDefaultsAndDisable with kDP_SetDefaults to fix keyboard disabled in some UEFI

diepeterpan opened the pull request:
Add boot_argn to fix keyboard disabled in some UEFI BIOS after a reboot. https://github.com/acidanthera/bugtracker/issues/1838

kbd_fixdisable=1

diepeterpan committed:
Add boot_argn to fix keyboard disabled in some UEFI

Add boot_argn to fix keyboard disabled in some UEFI BIOS after a reboot. https://github.com/acidanthera/bugtracker/issues/1838

andreszerocross commented:
Can you add another method instead boot-args?? Like add properties to Keyboard device in ACPI like other voodooPS2Controller patch

diepeterpan commented:
> Can you add another method instead boot-args?? Like add properties to Keyboard device in ACPI like other voodooPS2Controller patch

 Nope, can’t do that. Anyway, I believe it’s easier vir users to add a boot arg in OC than creating an ACPI patch.

averycblack commented on VoodooPS2Keyboard/VoodooPS2Keyboard.cpp:
```diff
@@ -1923,7 +1923,12 @@ void ApplePS2Keyboard::setKeyboardEnable(bool enable)
-    request.commands[0].inOrOut = enable ? kDP_Enable : kDP_SetDefaultsAndDisable;
+    
+    if(_kbd_fixdisable)
+        request.commands[0].inOrOut = enable ? kDP_Enable : kDP_SetDefaults;
```
Looking at Linux (https://github.com/torvalds/linux/blob/master/drivers/input/keyboard/atkbd.c#L913), it should be fine to just leave it as SetDefaults. I think it'd be fine to just replace with `kDP_SetDefaults` or to at least have the fix enabled by default.

diepeterpan replied:
Did not want to change the default behavior of the driver; hence only those with the problem need to set the boot argn.

averycblack replied:
I think it’s fine to change the default behavior. Basically every device I’ve seen is affected by this bug, and the new behavior is what Linux does so it should be fine. Could merge it in and see if there are any reports of this breaking.

diepeterpan replied:
Done.

"I think it'd be fine to just replace with kDP_SetDefaults."

diepeterpan replied:
Oops, I spot code to remove, hang on.

diepeterpan committed:
Replace kDP_SetDefaultsAndDisable with kDP_SetDefaults to fix keyboard disabled in OC on reboot.

Replace kDP_SetDefaultsAndDisable with kDP_SetDefaults to fix keyboard disabled in OC on reboot.

diepeterpan committed:
Update VoodooPS2Keyboard.cpp

Remove redundant code.

diepeterpan reviewed (commented):
Should now be ready to merge.

andreszerocross commented:
It will be better to add this new boot argument in README.md, IMHO

averycblack commented:
There is no new boot argument. SetDefaultsAndDisable was just replaced with SetDefaults.
